sqlalchemy_continuum/dialects/postgresql.py

```diff
@@ -51,13 +51,10 @@
 CREATE OR REPLACE FUNCTION {procedure_name}() RETURNS TRIGGER AS $$
 DECLARE transaction_id_value INT;
 BEGIN
-    BEGIN
-        transaction_id_value = (SELECT id FROM temporary_transaction);
-    EXCEPTION WHEN others THEN
-        RETURN NEW;
-    END;
+    transaction_id_value = (SELECT max(id) FROM temporary_transaction);
+
     IF transaction_id_value IS NULL THEN
-        RETURN NEW;
+        RAISE EXCEPTION 'Could not find latest transaction ID';
```

Comment:
This used to just silently not write to the applicable version table on error.  That's really bad for atomicity and sanctity of auditing, so I changed it

Reply:
What is/are the relevant failure mode(s) for us that this is now protecting us from? Can you add a comment accordingly?

Reply:
I think the error message could be a little more descriptive, and even provide a HINT: to make it clearer.

The only thing (and most important thing, tbh) this guards against is doing a write without an audit_transactions row present.

With this library, in the new world, ALL db transactions that modify tables that have versioning on must start with:
```
begin;
insert into audit_transactions(...) values (...);
```
In order to allow versioning.

If a postgres transaction doesn't have that insert statement at the start, then `transaction_id_value` will be null, which will cause that exception to be thrown.  So really the message should be `A database transaction must be created and a row must have been written to audit_transactions (or whatever your txtable is named) in order to write version history.` or _something_ along those lines.

I'll figure out a better message

Reply:
I think this enforcement makes sense at the DB-level. At the code level, I wonder how clear it will be to developers that you need to do this, and there are some situations where you could potentially have your "writes" pretty far from the transaction boundary. How would we enforce that the developer properly triggered the bump to the `audit_transactions` table in code?

Reply:
uhh continuum creates the audit_transactions row as a hook `after_begin`, I added that in this PR.  Here's the place: https://github.com/trialspark/sqlalchemy-continuum/pull/14/files#diff-6360ab8ee090093db566fe43480c5ff4R165

So doing this through application code/sqlalchemy, a developer should never even have to think about this.  But while doing raw SQL mutation in some database with triggers on, they will have to think about it

Reply:
I changed the message here to be far more descriptive!